e2e/staking-pool.e2e.ts

```diff
@@ -0,0 +1,236 @@
+        value: rewards,
+    });
+
+    setChainConfig(chainId, { stakingPoolFactoryAddress: stakingPoolFactory.address });
+};
+
+const putStake = async (pool: StakingPoolService, amount: number | BigNumber, withAmountCheck = true) => {
+    await pool.putStake(amount);
+
+    const currentStake = await pool.getStake();
+
+    if (withAmountCheck) {
+        expect(currentStake.amount).toStrictEqual(amount);
+    }
+
+    expect(currentStake.status).toBe(StakeStatus.STAKING);
+
+    const reward = await pool.checkReward();
+    expect(reward).toStrictEqual(zeroEwt);
+};
+
+const timeTravel = async (provider: providers.JsonRpcProvider, seconds: number) => {
+    await provider.send("evm_increaseTime", [seconds]);
+    await provider.send("evm_mine", []);
+};
+
+describe("StakingPool tests", () => {
+    // let domainsService: DomainsService;
+    let signerService: SignerService;
+    let stakingPoolService: StakingFactoryService;
+
+    beforeAll(async () => {
+        await replenish(orgOwner.address);
+        await replenish(patron.address);
+        await replenish(rootOwner.address);
+        await setupENS(rootOwner.address);
+        await setupStakingPoolFactory();
+        let connectToCacheServer;
+        ({ connectToCacheServer, signerService } = await initWithPrivateKeySigner(rootOwner.privateKey, rpcUrl));
+
+        const { stakingPoolService: stakingPools } = await connectToCacheServer();
+        stakingPoolService = stakingPools;
+    });
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        await setupStakingPoolFactory();
+        await signerService.connect(patron, ProviderType.PrivateKey);
+
+        await replenish(patron.address);
+        await timeTravel(provider, 20);
+    });
+
+    it("can stake", async () => {
+        const pool = await stakingPoolService.getPool();
+
+        await putStake(pool, oneEWT);
+    });
+
+    it("should reward", async () => {
+        const pool = await stakingPoolService.getPool();
+        await putStake(pool, oneEWT);
+
+        const periods = 24 * 90; // 3 months
+        const hour = 3600;
+
+        await timeTravel(provider, hour * periods);
+
+        const rewardAfterTimeTravel = await pool.checkReward();
+
+        expect(rewardAfterTimeTravel.gt(zeroEwt)).toBe(true);
+    });
+
+    it("should return all staked", async () => {
+        const pool = await stakingPoolService.getPool();
+
+        await putStake(pool, oneEWT);
+        await putStake(pool, oneEWT, false);
+
+        const currentStake = await pool.getStake();
+        expect(currentStake.amount).toStrictEqual(oneEWT.mul(2));
+
+        await pool.withdraw();
+        const stakeAfterWithdraw = await pool.getStake();
+
+        expect(stakeAfterWithdraw.amount).toStrictEqual(zeroEwt);
+
+        const { status } = await pool.getStake();
+        expect(status).toBe(StakeStatus.NONSTAKING);
+    });
+
+    it("should withdraw without reward", async () => {
```

Comment:
If this should test withdraw is possible when not staking, then there is no need to put stake before withdraw